Person#toString(): update result message

jonahtanjz opened the pull request:
Fixes #453.

Update result message with ";" as delimiters and omit "Tags:" if it is not present.

damithc commented:
@Zhiyuan-Amos any chance you can review this PR too?

Zhiyuan-Amos commented:
Hi @jonahtanjz! Thanks for submitting this PR, and I'm just quickly reviewing this. Based on our [Commit Organization Convention](https://oss-generic.github.io/process/docs/FormatsAndConventions.html#commit-organization):

> Each commit contains a single logical change

Your commit consists of 2 changes, namely:
1. Remove "Tags:" when user did not input any tags.
2. Add delimiter.

So you should break up your changes into 2 separate commits. To do so, you might have to familiarise yourself with some of the more advanced git commands such as `git reset`. :)

---

Sidenote: Your changes could have been accepted already if it's for other open source projects (or maybe even at work, depending on the engineering practices of the company), but this project is more particular about coding standards etc because it's used by students to learn Software Engineering :)

Zhiyuan-Amos commented on src/main/java/seedu/address/model/person/Person.java:
```diff
@@ -106,13 +106,16 @@ public int hashCode() {
+                .append("; Address: ")
+                .append(getAddress());
+
+        if (getTags().size() > 0) {
```
1. Assign `getTags` to a variable. See https://github.com/se-edu/addressbook-level4/pull/1026#discussion_r535838362. Also, in this scenario, the code will be creating an unmodifiable wrapper of `tags` twice, which is unnecessary.

1. Can we use `isEmpty()` instead? It's a more concise version of `getTags().size() > 0`.

jonahtanjz replied:
Alright thanks. I've made the changes.

jonahtanjz committed:
Person: Add delimiters to result message

There are no delimiters between each field entered.

The result message is not as user-friendly.

Let's add delimiters to the result message.

jonahtanjz committed:
Person: Remove 'Tags: ' if there are no tags

An additional phrase 'Tags:' is always present even if the user did
not input any tags.

This is not intuitive as the result message is displaying
unnecessary information.

Let's remove 'Tags:' from the result message if tags are not present.

jonahtanjz commented:
Thanks @damithc and @Zhiyuan-Amos for the comments. I've separated the commits and made the suggested changes.

Zhiyuan-Amos commented on src/main/java/seedu/address/model/person/Person.java:
```diff
@@ -111,9 +111,13 @@ public String toString() {
+        if (!tags.isEmpty()) {
+            builder.append("; Tags: ");
+        }
+        tags.forEach(builder::append);
```
I think the following is more intuitive:

```
Set<Tag> tags = getTags();
if (!tags.isEmpty()) {
    builder.append("; Tags: ");
    tags.forEach(builder::append);
}
```

Because it can be understood as: "If tags exist, add the header `; Tags: ` followed by the corresponding tags." Rather than calling `tags.forEach(builder::append);` regardless of whether the tags exist.

What do you think? :)

jonahtanjz replied:
Yes this seems a lot more intuitive as `tags.forEach(builder::append);` should not be called if `tags` is empty.

Zhiyuan-Amos reviewed (changes requested):
Few more comments:

1. There should be another newline after the commit title. Based on the post by `canihasreview` https://github.com/se-edu/addressbook-level4/pull/1028#issuecomment-743001299, you should only be seeing:

```
[v2 1/2] Person: Add delimiters to result message
[v2 2/2] Person: Remove 'Tags: ' if there are no tags
```

2. For the 2nd paragraph of your commit messages, remove the phrase `As a result`. Based on our commit organisation docs, it's already known that the 2nd paragraph is about the consequences of the 1st paragraph, so there's no need for this phrase.

You can amend commit messages using interactive rebase and force pushing afterwards.

---

Just some minor nits before we can have this merged!

j-lum commented:
@jonahtanjz Remember to submit a new iteration when you're ready.

jonahtanjz commented:
Thanks @Zhiyuan-Amos. I have updated the commit message and made the changes.
Thanks @j-lum for the reminder.

Zhiyuan-Amos reviewed (approved):
LGTM! @damithc

damithc commented:
Thanks @jonahtanjz for the PR and @Zhiyuan-Amos for the meticulous review. 👍
